pkg/crd-schema/gen.go

```diff
@@ -0,0 +1,352 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crdschema
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+
+	crdgen "sigs.k8s.io/controller-tools/pkg/crd"
+	crdschemayaml "sigs.k8s.io/controller-tools/pkg/crd-schema/yaml"
+	crdmarkers "sigs.k8s.io/controller-tools/pkg/crd/markers"
+	"sigs.k8s.io/controller-tools/pkg/genall"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+)
+
+var (
+	scheme = runtime.NewScheme()
+	codecs = serializer.NewCodecFactory(scheme)
+)
+
+func init() {
+	utilruntime.Must(apiextensionsv1beta1.AddToScheme(scheme))
+}
+
+// Generator is a genall.Generator that generates CRD schemas.
+type Generator struct {
+	// ManifestPath is a path with CustomResourceDefinition manifest files
+	ManifestPath string
+	// If true, no file is written. The Generate func will return an
+	// error if a file on disk differs from the generated manifest.
+	VerifyOnly bool
```

Comment:
nit: verifyonly implies that verification is always done.  probably want a different name.

Reply:
no. Verify-only is what we have in kube iterators: it is dry-run plus rc!=0 for changes.

Reply:
Is there an equivalent in the controller-gen cli framework?